src/main/java/mate/academy/dao/daoImpl/BookDaoImpl.java

```diff
@@ -0,0 +1,119 @@
+package mate.academy.dao.daoImpl;
+
+import mate.academy.exceptions.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Book;
+import mate.academy.dao.BookDao;
+import mate.academy.services.ConnectionUtil;
+import java.sql.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+@Dao
+public class BookDaoImpl implements BookDao {
+    @Override
+    public Book create(Book book) {
+        String query = "INSERT INTO BOOK (book_id, title, author," +
+                " publication_year, genre, price)" +
+                "VALUE (?, ?, ?, ?, ?, ?)";
+
+        try (Connection connection = ConnectionUtil.getConnection();
+             PreparedStatement statement =
+                     connection.prepareStatement(
+                             query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setLong(1, book.getBook_id());
+            statement.setString(2, book.getTitle());
+            statement.setString(3, book.getAuthor());
+            statement.setInt(4, book.getPublication_year());
+            statement.setString(5, book.getGenre());
+            statement.setDouble(6, book.getPrice());
```

Comment:
code is duplicated in update method, please introduce separate private method and reuse it in both cases